### DISCUSSION TITLE
grammar isn't found during runtime

### DISCUSSION THREAD

**Discussion by D0zee** [eclipse/tm4e#570] [Q&A] [answered]
Hello everyone! I'm trying to connect tm4e to my plugin. I created [simplest example of text mate grammar](https://pastebin.com/E8AzdWKd) and tried to connect this grammar in plugin.xml by [these lines of code](https://pastebin.com/n9QmbQbj). But when I try to launch my plugin, I get exception from tm4e like [No grammar provided for <source.test>]. I place test.tmLanguage.json inside src/syntaxes, I even added src in binary build and source build. It didn't help. Maybe I do something wrong? 

my build.properties looks like [this](https://pastebin.com/8DquR8Ty)

**Comment by sebthom** [comment, marked as answer]
does this work for you?

```xml
<extension point="org.eclipse.core.contenttype.contentTypes">
  <content-type
      base-type="org.eclipse.core.runtime.text"
      file-extensions="properties"
      id="tm4.test"
      name="Test file"
      priority="normal" />
  <file-association content-type="tm4.test" file-extensions="properties" />
</extension>

<extension point="org.eclipse.tm4e.registry.grammars">
  <grammar path="src/syntaxes/test.tmLanguage.json" scopeName="source.test" />
  <scopeNameContentTypeBinding contentTypeId="tm4.test" scopeName="source.test" />
</extension>

<extension point="org.eclipse.ui.editors">
  <editorContentTypeBinding 
      editorId="org.eclipse.ui.genericeditor.GenericEditor"
      contentTypeId="tm4.test" />
</extension>
```

**Reply by D0zee** [reply]
Unfortunately, I have the same exception: full stack trace is [here](https://pastebin.com/ddUdWm5Z)

I collect all files to repository. Link is [here](https://github.com/D0zee/test)

**Reply by D0zee** [reply]
Error was in that I don't define scopeName inside json with grammar. Thank you for answer!

**Reply by mickaelistria** [reply]
@D0zee Thanks for the followup! If you think this is something TM4E can do better (ie supporting grammars without scopeName, if this is a common case), feel free to submit a PR to improve that.
